include/seastar/core/semaphore.hh

```diff
@@ -488,6 +540,7 @@ public:
         if (units > _n) {
             throw std::invalid_argument("Cannot take more units than those protected by the semaphore");
         }
+        _sem->sub_outstanding_units(_n);
```

Comment:
split doesn't change units held

Reply:
ah, but the constructor will re-add them

Reply:
yes

Reply:
but there's a typo here.  need to s/_n/units/. will fix in next version